pkg/operation/botanist/component/vpa/exporter.go

```diff
@@ -0,0 +1,172 @@
+}
+
+func (v *vpa) reconcileExporterServiceAccount(serviceAccount *corev1.ServiceAccount) {
+	serviceAccount.AutomountServiceAccountToken = pointer.Bool(false)
+}
+
+func (v *vpa) reconcileExporterClusterRole(clusterRole *rbacv1.ClusterRole) {
+	clusterRole.Labels = getRoleLabel()
+	clusterRole.Rules = []rbacv1.PolicyRule{{
+		APIGroups: []string{"autoscaling.k8s.io"},
+		Resources: []string{"verticalpodautoscalers"},
+		Verbs:     []string{"get", "watch", "list"},
+	}}
+}
+
+func (v *vpa) reconcileExporterClusterRoleBinding(clusterRoleBinding *rbacv1.ClusterRoleBinding, clusterRole *rbacv1.ClusterRole, serviceAccount *corev1.ServiceAccount) {
+	clusterRoleBinding.Labels = getRoleLabel()
+	clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "ClusterRole",
+		Name:     clusterRole.Name,
+	}
+	clusterRoleBinding.Subjects = []rbacv1.Subject{{
+		Kind:      rbacv1.ServiceAccountKind,
+		Name:      serviceAccount.Name,
+		Namespace: serviceAccount.Namespace,
+	}}
+}
+
+func (v *vpa) reconcileExporterDeployment(deployment *appsv1.Deployment, serviceAccount *corev1.ServiceAccount) {
+	deployment.Labels = getAllLabels(exporter)
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas:             pointer.Int32(1),
+		RevisionHistoryLimit: pointer.Int32(2),
+		Selector:             &metav1.LabelSelector{MatchLabels: getAppLabel(exporter)},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: getAllLabels(exporter),
+			},
+			Spec: corev1.PodSpec{
+				ServiceAccountName: serviceAccount.Name,
+				Containers: []corev1.Container{{
+					Name:            "exporter",
+					Image:           v.values.Exporter.Image,
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					Command: []string{
+						"/usr/local/bin/vpa-exporter",
+						fmt.Sprintf("--port=%d", exporterPortMetrics),
+					},
+					Ports: []corev1.ContainerPort{{
+						Name:          "metrics",
+						ContainerPort: exporterPortMetrics,
+						Protocol:      corev1.ProtocolTCP,
+					}},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("30m"),
+							corev1.ResourceMemory: resource.MustParse("50Mi"),
+						},
+						Limits: corev1.ResourceList{
+							corev1.ResourceMemory: resource.MustParse("200Mi"),
+						},
+					},
+				}},
+			},
+		},
+	}
+}
+
+func (v *vpa) reconcileExporterVPA(vpa *vpaautoscalingv1.VerticalPodAutoscaler, deployment *appsv1.Deployment) {
+	updateMode := vpaautoscalingv1.UpdateModeAuto
+
+	vpa.Spec = vpaautoscalingv1.VerticalPodAutoscalerSpec{
+		TargetRef: &autoscalingv1.CrossVersionObjectReference{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+			Name:       deployment.Name,
+		},
+		UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{UpdateMode: &updateMode},
+		ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
```

Comment:
So far, when we refactored Helm charts and saw VPA objects without `resourcePolicy` we always added it (since we touched them anyways). I just followed this "paradigm" here and I think it's a nice improvement we're getting on the way.